src/Chapter2.hs

```diff
@@ -857,7 +896,9 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind [] = []
+rewind l = last l : rewind (init l)
```

Comment:
Your solution is correct! However, in lists, it is quite slow to add anything at the end of the list, as well as taking elements from the list end. That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go

Reply:
Aha, `go` patterns look awesome!

I think I got it right, can you check? Thanks!